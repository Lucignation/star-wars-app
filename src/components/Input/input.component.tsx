import { FC, useState, useEffect, ChangeEvent } from 'react';

import styles from './input.module.scss';

type props = {
  formType?: string;
  placeholder?: string;
  value: string | number;
  setValue: any;
  name: string;
};

const Input: FC<props> = ({ formType, placeholder, name, setValue, value }) => {
  const [showPassword, setShowPassword] = useState<boolean>(false);

  //   const onChange = (e: ChangeEvent<HTMLInputElement>) => {
  //     setValue(e.target.value);
  //   };

  return (
    <div className={styles.input_container}>
      {formType === 'password' ? (
        <div className={`${styles.singleInput}`}>
          <input
            type={showPassword ? 'text' : 'password'}
            className={`${styles.input}`}
            name={name}
            value={value}
            onChange={(e) => setValue(e.target.value)}
          />
          <small className={styles.placeholder}>{placeholder}</small>
        </div>
      ) : (
        <div className={`${styles.singleInput}`}>
          <input
            type='text'
            className={`${styles.input}`}
            name={name}
            value={value}
            onChange={(e) => setValue(e.target.value)}
          />
          <small className={styles.placeholder}>{placeholder}</small>
        </div>
      )}
    </div>
  );
};

export default Input;
