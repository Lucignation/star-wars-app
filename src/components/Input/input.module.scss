.singleInput {
  margin-bottom: 30px;
  position: relative;
}

.input {
  padding: 16px;
  font-family: 'Inter', sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  box-sizing: border-box;
  //   line-height: 24px;

  /* Text 1 */

  color: #0b2253;
  width: 335px;
  border-radius: 4px;
  border: 1px solid #a4a7b7;
}

.placeholder {
  position: absolute;
  top: -8px;
  left: 10px;
  background-color: #fff;
}

.input:focus {
  outline: #0a74dc;
  border: 1px solid #0a74dc;
}
