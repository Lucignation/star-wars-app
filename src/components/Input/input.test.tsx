import Input from './input.component';
import renderer from 'react-test-renderer';

const mockProps = {
  formType: 'text',
  placeholder: 'First Name',
  value: 'Gerald',
  setValue: () => 'Gerald',
  name: 'firstName',
};

describe('Input: ', () => {
  it('renders a snapshot of <Input /> when input is passed to a component ', () => {
    const input = renderer.create(<Input {...mockProps} />).toJSON();

    expect(input).toMatchSnapshot();
  });
});
