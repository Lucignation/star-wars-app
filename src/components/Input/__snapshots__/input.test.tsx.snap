// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input:  renders a snapshot of <Input /> when input is passed to a component  1`] = `
<div
  className="input_container"
>
  <div
    className="singleInput"
  >
    <input
      className="input"
      name="firstName"
      onChange={[Function]}
      type="text"
      value="Gerald"
    />
    <small
      className="placeholder"
    >
      First Name
    </small>
  </div>
</div>
`;
